src/Sentry/Internal/BatchProcessor.cs

```diff
@@ -0,0 +1,93 @@
+using System.Timers;
+using Sentry.Protocol;
+using Sentry.Protocol.Envelopes;
+
+namespace Sentry.Internal;
+
+/// <summary>
+/// The Sentry Batch Processor.
+/// This implementation is not complete yet.
+/// Also, the specification is still work in progress.
+/// </summary>
+/// <remarks>
+/// Sentry Specification: <see href="https://develop.sentry.dev/sdk/telemetry/spans/batch-processor/"/>.
+/// OpenTelemetry spec: <see href="https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/batchprocessor/README.md"/>.
+/// </remarks>
+internal sealed class BatchProcessor : IDisposable
+{
+    private readonly IHub _hub;
+    private readonly BatchProcessorTimer _timer;
+    private readonly BatchBuffer<SentryLog> _logs;
+    private readonly object _lock;
+
+    private DateTime _lastFlush = DateTime.MinValue;
+
+    public BatchProcessor(IHub hub, int batchCount, TimeSpan batchInterval)
+        : this(hub, batchCount, new TimersBatchProcessorTimer(batchInterval))
+    {
+    }
+
+    public BatchProcessor(IHub hub, int batchCount, BatchProcessorTimer timer)
+    {
+        _hub = hub;
+
+        _timer = timer;
+        _timer.Elapsed += OnIntervalElapsed;
```

Comment:
are going to have more than one subscriber?
I'd avoid events. It's not great with asynchronous code (`async void`), if one subscribers throws the second might not get the message etc.

How about a Func or Action?